core/src/main/java/org/elasticsearch/discovery/zen/publish/PublishClusterStateAction.java

```diff
@@ -407,7 +412,7 @@ void validateIncomingState(ClusterState incomingState, ClusterState lastSeenClus
             final String message = String.format(
```

Comment:
With the new check (based on the ClusterService.state()), a cluster state with a lower/equal version than the current state will be rejected. If it has a higher version it will get in. If it is lower than the last seen cluster state it means that it will be cleaned when the first of these happen:
1) A commit message for this specific CS comes in.
2) A commit message for a higher CS comes in (including the lastSeenClusterState) and that CS is processed.

Is there a case where the lastSeenClusterState.supersedes based protection helps and isn't covered by the above?